test/testenv/cmutil.go

```diff
@@ -117,24 +117,95 @@ type ClusterMasterSearchHeadContent struct {
 	Status       string      `json:"status"`
 }
 
-// CheckSearchHeadRemoved check if search head is removed from Indexer Cluster
-func CheckSearchHeadRemoved(deployment *Deployment) bool {
+// ClusterMasterPeersResponse /services/cluster/master/peers response
+type ClusterMasterPeersResponse struct {
+	Entry []struct {
+		Content struct {
+			Label                                  string `json:"label"`
+			RegisterSearchAddress                  string `json:"register_search_address"`
+			ReplicationCount                       int    `json:"replication_count"`
+			ReplicationPort                        int    `json:"replication_port"`
+			ReplicationUseSsl                      bool   `json:"replication_use_ssl"`
+			RestartRequiredForApplyingDryRunBundle bool   `json:"restart_required_for_applying_dry_run_bundle"`
+			Site                                   string `json:"site"`
+			SplunkVersion                          string `json:"splunk_version"`
+			Status                                 string `json:"status"`
+			SummaryReplicationCount                int    `json:"summary_replication_count"`
+		} `json:"content"`
+	} `json:"entry"`
+}
+
+// GetIndexersOnCM get indexers on Cluster Master
+func GetIndexersOnCM(deployment *Deployment) ClusterMasterSearchHeadResponse {
 	//code to execute
 	podName := fmt.Sprintf("splunk-%s-cluster-master-0", deployment.GetName())
-	stdin := "curl -ks -u admin:$(cat /mnt/splunk-secrets/password) https://localhost:8089/services/cluster/master/searchheads?output_mode=json"
+	stdin := "curl -ks -u admin:$(cat /mnt/splunk-secrets/password) https://localhost:8089/services/cluster/master/peers?output_mode=json"
 	command := []string{"/bin/sh"}
 	stdout, stderr, err := deployment.PodExecCommand(podName, command, stdin, false)
+	restResponse := ClusterMasterSearchHeadResponse{}
 	if err != nil {
 		logf.Log.Error(err, "Failed to execute command on pod", "pod", podName, "command", command)
-		return false
+		return restResponse
 	}
 	logf.Log.Info("Command executed on pod", "pod", podName, "command", command, "stdin", stdin, "stdout", stdout, "stderr", stderr)
+	err = json.Unmarshal([]byte(stdout), &restResponse)
+	if err != nil {
+		logf.Log.Error(err, "Failed to parse cluster peers")
+	}
+	return restResponse
+}
+
+// CheckIndexerOnCM check given Indexer on cluster master
+func CheckIndexerOnCM(deployment *Deployment, indexerName string) bool {
+	restResponse := GetIndexersOnCM(deployment)
+	found := false
+	for _, entry := range restResponse.Entries {
+		logf.Log.Info("Peer found On CM", "Indexer Name", entry.Content.Label, "Status", entry.Content.Status)
+		if entry.Content.Label == indexerName {
+			found = true
+			break
+		}
+	}
+	return found
+}
+
+// GetSearchHeadsOnCM get search head listed on Cluster Master
+func GetSearchHeadsOnCM(deployment *Deployment) ClusterMasterSearchHeadResponse {
```

Comment:
NIT: can we combined into a single function with the 'api' value.... i.e searchheads or peers passed as a parameter

Reply:
The reason I kept it seperate was I expected different response back from API. However revisiting it can be combined as the response content is similar.